### PR TITLE
docs: Improve structure of user guide

### PR DESCRIPTION
**Pull request by r-brink**
Draft PR to track the progress on this issue https://github.com/pola-rs/polars/issues/13626

Changes so far:

-  remove the 'homepage'
-  enter on user-guide/overview page
- remove basics session and create single getting started page

- more straightforward getting-started page
- adding index pages to user guide for reference

### CODE REVIEW

**Comment by r-brink** [comment]
@c-peters can you have a look?

File: docs/user-guide/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+Polars supports reading and writing to all common files (e.g. csv, json, parquet), cloud storage (S3, Azure Blob, BigQuery) and databases (e.g. postgres, mysql). Below we use csv as example to demonstrate foundational read/write operations.
+
+{{code_block('user-guide/basics/reading-writing','dataframe',['DataFrame'])}}
+
+```python exec="on" result="text" session="getting-started/reading"
+--8<-- "python/user-guide/basics/reading-writing.py:dataframe"
+```
+
+### CSV example
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:33]: This does not need to be a separate header, maybe we can remove it altogether

File: docs/user-guide/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+In this example we write the DataFrame to `output.csv`. After that we can read it back with `read_csv` and `print` the result for inspection.
+
+{{code_block('user-guide/basics/reading-writing','csv',['read_csv','write_csv'])}}
+
+```python exec="on" result="text" session="getting-started/reading"
+--8<-- "python/user-guide/basics/reading-writing.py:csv"
+```
+
+For more examples on the CSV file format and other data formats, start here [IO section on CSV](io/csv.md) of the User Guide.
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:43]: We can now link to the index page of IO

File: docs/user-guide/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+{{code_block('user-guide/basics/reading-writing','dataframe',['DataFrame'])}}
+
+```python exec="on" result="text" session="getting-started/reading"
+--8<-- "python/user-guide/basics/reading-writing.py:dataframe"
+```
+
+### CSV example
+
+In this example we write the DataFrame to `output.csv`. After that we can read it back with `read_csv` and `print` the result for inspection.
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:35]: Duplicate with `Below we use csv as example to demonstrate foundational read/write operations.`

File: docs/user-guide/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    cargo add polars -F lazy
+
+    # Or Cargo.toml
+    [dependencies]
+    polars = { version = "x", features = ["lazy", ...]}
+    ```
+
+## Reading & writing
+
+Polars supports reading and writing to all common files (e.g. csv, json, parquet), cloud storage (S3, Azure Blob, BigQuery) and databases (e.g. postgres, mysql). Below we use csv as example to demonstrate foundational read/write operations.
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:25]: foundational read/write operations => the concept of reading and writing to disk

File: docs/user-guide/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+```python exec="on" result="text" session="getting-started/reading"
+--8<-- "python/user-guide/basics/reading-writing.py:csv"
+```
+
+For more examples on the CSV file format and other data formats, start here [IO section on CSV](io/csv.md) of the User Guide. 
+
+## Expressions
+
+`Expressions` are the core strength of Polars. The `expressions` offer a versatile structure that both solves easy queries and is easily extended to complex ones. Below we cover the basic components that serve as building block (or in Polars terminology contexts) for all your queries:
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:47]: versatile structure that both solves easy queries and is easily extended to complex ones =>
modular structure that allows you to combine simple concepts into complex queries

File: docs/user-guide/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+- `select`
+- `filter`
+- `with_columns`
+- `group_by`
+
+To learn more about expressions and the context in which they operate, see the User Guide sections: [Contexts](concepts/contexts.md) and [Expressions](concepts/expressions.md).
+
+### Select statement
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:56]: Select Statement => Select

File: docs/user-guide/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+{{code_block('user-guide/basics/expressions','filter2',['filter'])}}
+
+```python exec="on" result="text" session="getting-started/expressions"
+print(
+    --8<-- "python/user-guide/basics/expressions.py:filter2"
+)
+```
+
+### With_columns
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:108]: With_columns => Add columns

File: docs/user-guide/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+{{code_block('user-guide/basics/expressions','group_by2',['group_by'])}}
+
+```python exec="on" result="text" session="getting-started/expressions"
+print(
+    --8<-- "python/user-guide/basics/expressions.py:group_by2"
+)
+```
+
+### Combining operations
```
Comment (c-peters) [inline docs/user-guide/getting-started.md:147]: Combining operation => Combination

**Review by c-peters — APPROVED** [review]

**Pull request merged**